hack/common.sh

```diff
@@ -645,7 +645,14 @@ function os::build::os_version_vars() {
     OS_GIT_SHORT_VERSION="${OS_GIT_COMMIT}"
 
     # Use git describe to find the version based on annotated tags.
-    if [[ -n ${OS_GIT_VERSION-} ]] || OS_GIT_VERSION=$("${git[@]}" describe "${OS_GIT_COMMIT}^{commit}" 2>/dev/null); then
+    if [[ -n ${OS_GIT_VERSION-} ]] || OS_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=7 "${OS_GIT_COMMIT}^{commit}" 2>/dev/null); then
+      # This translates the "git describe" to an actual semver.org
+      # compatible semantic version that looks something like this:
+      #   v1.1.0-alpha.0.6+84c76d1142ea4d
```

Comment:
Yeah, although anyone tagging to a specific git commit won't be broken.

On Thu, Jul 7, 2016 at 3:41 PM, David Eads <notifications@github.com> wrote:

> In hack/common.sh
> <https://github.com/openshift/origin/pull/9578#discussion_r69973027>:
>
> > @@ -645,7 +645,14 @@ function os::build::os_version_vars() {
> >      OS_GIT_SHORT_VERSION="${OS_GIT_COMMIT}"
> >
> >      # Use git describe to find the version based on annotated tags.
> > -    if [[ -n ${OS_GIT_VERSION-} ]] || OS_GIT_VERSION=$("${git[@]}" describe "${OS_GIT_COMMIT}^{commit}" 2>/dev/null); then
> > +    if [[ -n ${OS_GIT_VERSION-} ]] || OS_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=7 "${OS_GIT_COMMIT}^{commit}" 2>/dev/null); then
> > +      # This translates the "git describe" to an actual semver.org
> > +      # compatible semantic version that looks something like this:
> > +      #   v1.1.0-alpha.0.6+84c76d1142ea4d
>
> This is going to affect how people can/should do their user-agent
> restrictions, right? Followup issue.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9578/files/cfedbc1708fc75ac73aba9d0e9e115b906cd8527..e3c71ab7b9d54bf3eb39c984f59d1decbf2b51f5#r69973027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pyG0owYP_7JA1SF6JXWvzsYF_3GZks5qTVZtgaJpZM4I-xTE>
> .
>